src/registry/addressNL/edit.stories.ts

```diff
@@ -0,0 +1,152 @@
+      title: 'Address NL',
+      icon: 'home',
+      group: 'basic',
+      weight: 10,
+      schema: {},
+    },
+  },
+} as Meta<typeof ComponentEditForm>;
+
+type Story = StoryObj<typeof ComponentEditForm>;
+
+export const PostcodeValidationTabWithoutConfiguration: Story = {
+  name: 'AddressNL postcode validation tab (no prior configuration)',
+  play: async ({canvasElement, step}) => {
+    const canvas = within(canvasElement);
+
+    await step('Navigate to validation tab and open Postcode validation', async () => {
+      await userEvent.click(canvas.getByRole('link', {name: 'Validation'}));
+      await userEvent.click(canvas.getAllByText('Postcode')[0]);
+      expect(await canvas.findByLabelText('Regular expression for postcode')).toBeVisible();
+      expect(await canvas.findByText('NL')).toBeVisible();
+      expect(await canvas.findByText('EN')).toBeVisible();
+      expect(await canvas.findByText('Error code')).toBeVisible();
+      expect(canvas.getByLabelText('Error message for "pattern"')).toHaveDisplayValue('');
+    });
+  },
+};
+
+export const CityValidationTabWithoutConfiguration: Story = {
+  name: 'AddressNL city validation tab (no prior configuration)',
+  play: async ({canvasElement, step}) => {
+    const canvas = within(canvasElement);
+
+    await step('Navigate to validation tab and open City validation', async () => {
+      await userEvent.click(canvas.getByRole('link', {name: 'Validation'}));
+      await userEvent.click(canvas.getAllByText('City')[0]);
+      expect(await canvas.findByLabelText('Regular expression for city')).toBeVisible();
+      expect(await canvas.findByText('NL')).toBeVisible();
+      expect(await canvas.findByText('EN')).toBeVisible();
+      expect(await canvas.findByText('Error code')).toBeVisible();
+      expect(canvas.getByLabelText('Error message for "pattern"')).toHaveDisplayValue('');
+    });
+  },
+};
+
+export const PostcodeValidationTabWithConfiguration: Story = {
+  name: 'AddressNL postcode validation tab (with prior configuration)',
+  args: {
+    component: {
+      id: 'wekruya',
+      type: 'addressNL',
+      key: 'address',
+      label: 'An address field',
+      openForms: {
+        components: {
+          postcode: {
+            validate: {
+              pattern: '1017 [A-Za-z]{2}',
+            },
+            translatedErrors: {
+              nl: {pattern: 'Postcode moet 1017 XX zijn'},
+              en: {pattern: 'Postal code must be 1017 XX'},
+            },
+          },
+        },
+      },
+    },
+  },
+  play: async ({canvasElement, step}) => {
+    const canvas = within(canvasElement);
+
+    await step('Navigate to validation tab and open Postcode validation', async () => {
+      await userEvent.click(canvas.getByRole('link', {name: 'Validation'}));
+      await userEvent.click(canvas.getAllByText('Postcode')[0]);
+      const patternInput = canvas.getByLabelText(
+        'Regular expression for postcode'
+      ) as HTMLInputElement;
```

Comment:
the `as` casting should not be needed, often you can pass the expected type as a generic param: `getByLabelText<HTMLInputElement>` *or* if the problem is that it can be `null` or `undefined`, you can add an exclamation mark at the end saying that you know it will resolve:

```tsx
canvas.getByLabelText("some label")!;
```

Reply:
I'll pass the generic param, the exclamation mark doesn't work, it gives me the same error as when I don't add the type at all: `Property 'value' does not exist on type 'HTMLElement'.ts(2339)` (on the line `expect(patternInput.value).toBe('Amsterdam');`)